website/content/functional-database.md

```diff
@@ -180,94 +180,246 @@ let res = curry arg3
+someone might say "why functional?" to which I say "exactly the question I was
+waiting for!".
+
+There are a few quick benefits I want to get out of the way:
+
+- Pipelines make the flow of the query "top-to-bottom". An earlier part of a the
+  pipeline will always be a valid query, regardless of what follows.
+- Functions on relations (transforms) are designed to be orthogonal. Each of
+  them has as few effects as possible and can be used at any point in the
+  pipeline (except for `from`).
+
+But there is more.
+
+### Aggregate
+
+At this point, I have to introduce `aggregate`. It takes a relation and produces
+a single row using some aggregation function:
+
+```prql
+(from albums | aggregate [n_albums = count])
+```
+
+What happens if we don't specify the leading relation?
+
+```prql
+(aggregate [n_albums = count])
+```
+
+Because `aggregate` is missing an argument, currying kicks in, and the whole
+expression evaluates to a function that is still waiting for a relation.
+
+In SQL it is common to use GROUP BY when aggregating. If you think about it, it
+essentially separates the relation into groups using some criteria and then
+applies `aggregate` to each of the groups. This is exactly how PRQL expressed
+it:
+
+```prql
+from albums | group artist_id (aggregate [n_albums = count])
+```
+
+This is a lot for one line, so let's unveil new syntactic conveniences: a new
+line is a pipe operator and the top-level pipeline does not need parenthesis.
+I'll also add a new transform at the back, don't worry about it.
+
+```prql
+from albums
+group artist_id (
+    aggregate [n_albums = count]
+)
+filter n_albums > 3
+```
+
+Notice how `group` operates on the whole relation it gets from `from` and that
+`aggregate` is passed to `group` as a function to be applied to each of the
+groups. `aggregate` then converts each of the groups into a single row and
+returns that to `group` that composes these rows back together. This can now be
+passed on to `filter`, which will remove any rows that don't match its
+condition.
+
+In SQL, a similar expression would have these four parts:
+
+- projection in SELECT,
+- an aggregation function that implicitly triggers aggregation,
+- GROUP BY clause,
+- HAVING clause.
+
+These parts are entangled syntactically and semantically into one feature we
+understand as GROUPING.
+
+The beautiful realization is that these are 3 different operations that are
+happening and that when separated they don't need to be associated with each
+other. For example `filter` is more commonly expressed as WHERE and `group` has
+other uses than `aggregate`.
+
+But to separate these core relational operations, we need a way to express
+aggregate as a function. This is why the functional paradigm fits the relational
+data model.
+
+### Distinct
+
+A common question when learning SQL is "how do I select the row where column x
+is min?". It has many variations, but there are two ways of doing it:
```

Comment:
I would say "where column x is **smallest**" or "where column x is **minimised**".